packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -232,6 +233,38 @@ describe("Garbage Collection Tests", () => {
 		);
 	});
 
+	it("Pending session expiry does not close container", () => {
+		let closeCalled = false;
+		const timenow = 5000; // arbitrary number
+		clock.tick(timenow);
+		gc = createGarbageCollector(
+			{ sessionExpiryTimerStarted: 0 },
+			undefined /* gcBlobsMap */,
+			undefined /* gcMetadata */,
+			() => {
+				closeCalled = true;
+			},
+		);
+		assert(closeCalled === false, "Close should have been called");
+	});
+
+	it("Pending session expiry closes container", () => {
+		let closeCalled = false;
+		const timethen = 12345;
+		const timenow = defaultSessionExpiryDurationMs;
+		clock.tick(timenow + timethen);
+		gc = createGarbageCollector(
+			{ sessionExpiryTimerStarted: timethen },
+			undefined /* gcBlobsMap */,
+			undefined /* gcMetadata */,
+			() => {
+				closeCalled = true;
+			},
+		);
+		clock.tick(1);
+		assert(closeCalled, "Close should have been called");
+	});
```

Comment:
Maybe add a test that does `assert.throws(() => createGarbageCollector(...))` (with the proper numbers passed so it's already too old), since we were so intentional about that decision.